worker-workflow/src/main/java/com/github/cafdataprocessing/WorkflowWorker.java

```diff
@@ -0,0 +1,203 @@
+        final TransformWorkflowResult transformWorkflowResult = transformWorkflow(extractedProperties, document);
+        if(transformWorkflowResult==null){
+            LOG.warn("Failure during workflow transformation. Processing of this document will not proceed " +
+                    "for this worker.");
+            return;
+        }
+
+        // Evaluate the document against the workflow to execute the first matching action. If document has to be sent to another
+        // worker its queue property will be updated to reflect this.
+        try
+        {
+            WorkflowEvaluator.evaluate(document, transformWorkflowResult.getTransformedWorkflow(),
+                    transformWorkflowResult.getWorkflowStorageRef());
+        } catch (final PostProcessingFailedException e) {
+            LOG.error("A failure occurred trying to evaluate document against transformed Workflow.", e);
+            document.addFailure(WorkflowWorkerConstants.ErrorCodes.WORKFLOW_EVALUATION_FAILED, e.getMessage());
+        }
+    }
+
+    /**
+     * Retrieves transformed workflow result based on extracted properties passed. If unable to retrieve a result then
+     * a failure will be added to the passed {@code document} and {@code null} returned.
+     * @param extractedProperties properties to use in workflow retrieval and transformation.
+     * @param document document to update with failure details in event of any issues.
+     * @return the transformed workflow result matching provided properties or null if there is a failure retrieving the
+     * transformed workflow result.
+     * @throws DocumentWorkerTransientException if there is a transient failure during workflow transformation.
+     */
+    private TransformWorkflowResult transformWorkflow(final ExtractedProperties extractedProperties, final Document document)
+            throws DocumentWorkerTransientException {
+        try {
+            try {
+                return workflowCache.getTransformWorkflowResult(extractedProperties.getWorkflowId(),
+                        extractedProperties.getProjectId(), extractedProperties.getOutputPartialReference());
+            } catch (final ApiException | UncheckedExecutionException firstException) {
+                // may have been a transient API issue, check if the API is healthy
+                try {
+                    final HealthStatus processingApiHealth = workflowAdminApi.healthCheck();
+                    if (!HealthStatus.StatusEnum.HEALTHY.equals(processingApiHealth.getStatus())) {
+                        // processing API is unhealthy, throw a transient exception
+                        LOG.info("Unable to transform workflow as processing API is unhealthy.");
+                        throw new DocumentWorkerTransientException(
+                                "Unable to transform workflow. Processing API is unhealthy.");
+                    }
+                } catch (final Exception secondException) {
```

Comment:
I don't like that this is catching `Exception`, but it's in the middle of the area you're currently changing so this might be going away.

Reply:
Yep this is no longer there.